test/known_issues/test-url-parse-conformance.js

```diff
@@ -0,0 +1,52 @@
+'use strict';
+
+const common = require('../common');
+const url = require('url');
+const assert = require('assert');
+const path = require('path');
+
+const tests = require(path.join(common.fixturesDir, 'url-tests.json'));
+
+var failed = 0;
+var attempted = 0;
+
+tests.forEach((test) => {
+  attempted++;
+  if (typeof test === 'string') return;
+  const href = url.resolve(test.base, test.input);
+  const parsed = url.parse(href);
+
+  var username, password;
+  try {
+    assert.equal(test.href, parsed.href);
+    assert.equal(test.protocol, parsed.protocol);
+    username = parsed.auth ? parsed.auth.split(':', 2)[0] : '';
+    password = parsed.auth ? parsed.auth.split(':', 2)[1] : '';
+    assert.equal(test.username, username);
+    assert.equal(test.password, password);
+    assert.equal(test.host, parsed.host);
+    assert.equal(test.hostname, parsed.hostname);
+    assert.equal(+test.port, +parsed.port);
+    assert.equal(test.pathname, parsed.pathname || '/');
+    assert.equal(test.search, parsed.search || '');
+    assert.equal(test.hash, parsed.hash || '');
+    console.log(`\u2713 - ${test.href}`);
+  } catch (err) {
+    console.log(`\u2717 - ${test.href} - ${err.message}`);
+    failed++;
+  }
+  console.log(`    Input:    ${test.input}`);
```

Comment:
No, it doesn't. I added that in largely to highlight the differences. I can
switch it to only output on failing tests.
On Mar 24, 2016 5:27 AM, "Colin Ihrig" <notifications@github.com> wrote:

> In test/known_issues/test-url-parse-conformance.js
> <https://github.com/nodejs/node/pull/5885#discussion_r57309357>:
>
> > +    username = parsed.auth ? parsed.auth.split(':', 2)[0] : '';
> > +    password = parsed.auth ? parsed.auth.split(':', 2)[1] : '';
> > +    assert.equal(test.username, username);
> > +    assert.equal(test.password, password);
> > +    assert.equal(test.host, parsed.host);
> > +    assert.equal(test.hostname, parsed.hostname);
> > +    assert.equal(+test.port, +parsed.port);
> > +    assert.equal(test.pathname, parsed.pathname || '/');
> > +    assert.equal(test.search, parsed.search || '');
> > +    assert.equal(test.hash, parsed.hash || '');
> > +    console.log(`\u2713 - ${test.href}`);
> > +  } catch (err) {
> > +    console.log(`\u2717 - ${test.href} - ${err.message}`);
> > +    failed++;
> > +  }
> > +  console.log(`    Input:    ${test.input}`);
>
> Does all of this need to be logged for every test? What if it was only
> printed for failing tests?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/nodejs/node/pull/5885/files/e8229bd3262164bd842f00ec09fe3e533666eec6#r57309357>
>